src/pudl/transform/ferc714.py

```diff
@@ -0,0 +1,749 @@
+    "HST": "Pacific/Honolulu",
+}
+"""Mapping between standardized time offset codes and canonical timezones."""
+
+MISSING_UTILITY_ID_EIA = {
+    307: 14354,  # PacifiCorp
+    295: 40229,  # Old Dominion Electric Cooperative
+    329: 39347,  # East Texas Electricity Cooperative
+}
+"""FERC respondents lacking EIA utility IDs that appear to exist."""
+
+RENAME_COLS = {
+    "respondent_id_ferc714": {
+        "respondent_id": "utility_id_ferc714",
+        "respondent_name": "utility_name_ferc714",
+        "eia_code": "utility_id_eia",
+    },
+    "pa_demand_hourly_ferc714": {
+        "report_yr": "report_year",
+        "plan_date": "report_date",
+        "respondent_id": "utility_id_ferc714",
+        "timezone": "utc_offset_code",
+    },
+    "pa_description_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+        "elec_util_name": "utility_name_ferc714",
+        "peak_summer": "peak_demand_summer_mw",
+        "peak_winter": "peak_demand_winter_mw",
+    },
+    "id_certification_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "ba_gen_plants_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "ba_demand_monthly_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "ba_net_energy_load_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "adjacent_bas_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "ba_interchange_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "ba_lambda_hourly_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "lambda_description_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "pa_demand_forecast_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+}
+
+##############################################################################
+# Internal helper functions.
+##############################################################################
+
+
+def _hours_to_ints(col_name):
+    """A macro to rename hourly demand columns."""
+    if re.match(r"^hour\d\d$", col_name):
+        return int(col_name[4:])
+    else:
+        return col_name
+
+
+def _standardize_offset_codes(df, tz_fixes):
+    """
+    Convert to standardized UTC offset abbreviations.
+
+    This function ensures that all of the 3-4 letter abbreviations used to
+    indicate a timestamp's localized offset from UTC are standardized, so that
+    they can be used to make the timestamps timezone aware. The standard
+    abbreviations we're using are:
+
+    "HST": Hawaii Standard Time
```

Comment:
are these codes <> names useful in the code somewhere? seems like a dictionary to store somewhere instead of in a doc string.

Reply:
I don't think so -- a lot of the work that I was doing in the timeseries cleanup was to get rid of these (not actually standardized) codes completely -- so now it's using the official (and readable) timezone names, like `America/Chicago` instead.